devicetypes/smartthings/vision-in-wall-2relays-switch.src/vision-in-wall-2relays-switch.groovy

```diff
@@ -0,0 +1,294 @@
+/**
+ *  Vision In-Wall 2Relays Switch (Models: ZL7435xx-5)
+ *
+ *  Author:
+ *    Lan, Kuo Wei 
+ *
+ *  Product Link: 
+ *    http://www.visionsecurity.com.tw/index.php?option=product&lang=en&task=pageinfo&id=335&belongid=334&index=0
+*/
+metadata {
+	definition(
+		name: "Vision In-Wall 2Relays Switch", 
+		namespace: "SmartThings", 
+		author: "Lan, Kuo Wei", 
+		ocfDeviceType: "oic.d.switch", 
+		mnmn: "SmartThings", 
+		vid: "generic-switch"
+	) {
+		capability "Actuator"
+		capability "Health Check"
+		capability "Refresh"
+		capability "Switch"
+		capability "Configuration"
+		// This DTH uses 2 switch endpoints. Parent DTH controls endpoint 1 so please use '1' at the end of deviceJoinName
+		// Child device (isComponent : false) representing endpoint 2 will substitute 1 with 2 for easier identification.
+		fingerprint manufacturer: "0109", prod: "2017", model: "171B", deviceJoinName: "Vision 2Relays Switch 1"
+		fingerprint manufacturer: "0109", prod: "2017", model: "171C", deviceJoinName: "Vision 2Relays Switch 1"
+	}
+
+	preferences {
+		parameterMap.each {
+			input (title: it.name, description: it.description, type: "paragraph", element: "paragraph")
+			switch(it.type) {
+				case "boolean":
+					input(type: "paragraph", element: "paragraph", description: "Option enabled: ${it.activeDescription}\n" +
+						"Option disabled: ${it.inactiveDescription}"
+					)
+					input(name: it.key, type: "boolean", title: "Enable", defaultValue: it.defaultValue == it.activeOption, required: false)
+					break
+					
+				case "enum":
+					input(name: it.key, title: "Select", type: "enum", options: it.values, defaultValue: it.defaultValue, required: false)
+					break
+			}	
+		}
+	}
+}
+
+def installed() {
+	def componentLabel
+	if (device.displayName.endsWith('1')) {
+		componentLabel = "${device.displayName[0..-2]}2"
+	} else {		
+		componentLabel = "$device.displayName 2"
+	}
+	try {
+		String dni = "${device.deviceNetworkId}-ep2"
+		addChildDevice("Z-Wave Binary Switch Endpoint", dni, device.hub.id,
+			[completedSetup: true, label: "${componentLabel}", isComponent: false])
+		log.debug "Endpoint 2 (Z-Wave Binary Switch Endpoint) added as $componentLabel"
+	} catch (e) {
+		log.warn "Failed to add endpoint 2 ($desc) as Z-Wave Binary Switch Endpoint - $e"
+	}
+	// Preferences template
+	state.currentPreferencesState = [:]

```

Comment:
Since there is only one device setting in this DTH please handle Your setting the same way it's done in "Z-Wave Siren" instead of using the mechanism developed for handling large sets of device preferences. The code will then be optimal.

Reply:
This deivce is like "Z-Wave Dual Switch", so I used this DTH to reference,
need use this to create Endpoint-2 to control device's switch-2.

 "Z-Wave Siren"'s DTH only create one device, it have child device create?

Reply:
I meant that You might control setting those parameters the same way it's done in "Z-Wave Siren"'s DTH.

Reply:
Could I use this structure? because we will have new Z-Wave 700 product realease,
they will use more parameters and will upgrade this DTH in the future.

Reply:
How many new parameters ?

Reply:
Our company plans to function of automatic timer shutdown and auto restore relay Power on.
It is expected that Configuration will increase about 6 parameter number,
since it is still under development, I can't know all parameter if OK,
need wait some time let this new product finish.